### ISSUE TITLE
Support for i18n

### ISSUE THREAD

**Issue by matheusgrieger** [codex-team/editor.js#751]
I'm trying to translate the editor tooltips and names to Portuguese, but the docs don't specify a way to do that, and in fact, it doesn't seem possible at the moment.

Can this be done?

**Comment by realmike33** [comment]
Is it, you'll have to write the code for it 💃

**Comment by matheusgrieger** [comment]
Is that a "not available, needs contribution"? 😆 

I'll see what I can do to help and add this feature.

**Comment by neSpecc** [comment]
@matheusgrieger for now you can specify the `title` and `icon` with the `toolbox` property of Editor Config like this:

```js
 const editor = new EditorJS({
  tools: {
    image: {
      class: ImageTool,
      toolbox: {
        icon: '<svg>my icon code</svg>',
        title: 'My Tool Title'
      }
    }
  }
});
```

We will describe this moment at the docs more carefully. Also, we will think about full localisation of all UI texts.

**Comment by lnalex** [comment]
Is someone in charge of i18n for editor.js ? 

There is some hardcoded value: https://github.com/codex-team/editor.js/blob/3382da100076b997946523c1587d26dff7159ea3/src/components/modules/toolbar/index.ts#L127

So it's impossible to translate.

**Comment by t-itzel** [comment]
@neSpecc 

> We will describe this moment at the docs more carefully. Also, we will think about full localisation of all UI texts.

Any progress here? I  like this editor and want to integrate it into a VueJS application but I need the editor to be translated to german.

Worst case scenario would be to "translate" everything manually using
```
tools: {
  image: {
    class: ImageTool,
    toolbox: {
      title: '<<<TRANSLATION>>>'
    }
  },
  ...
}
```

even though, documentation is sparse when it comes to details about where to retrieve reference to be used in e.g. `class: ImageTool` from. I need some sort of import statement.

**Comment by neSpecc** [comment]
@t-itzel estimated 1-2 month. If you can contribute, it would be great and we would release it faster.

**Comment by t-itzel** [comment]
> @t-itzel estimated 1-2 months. If you can contribute, it would be great and we would release it faster.

alright, thanks. Sorry but I can't manage to contribute right now.

**Comment by ErriourRU** [comment]
Nothing has changed yet... Hard-coded phrases are in the kernel.

**Comment by blindibrasil** [comment]
How is the translation process going?
I would like to know if this support already exists, or if you are still planning to have it inserted in the next updates?
When released, I will be on hand to help with the translation into my language (PT-BR).
Thanks.

**Comment by glauberm** [comment]
@neSpecc anything I could do to help with the i18n?

**Comment by neSpecc** [comment]
@glauberm This task turned out to be more complex architecturally, so I've started experimenting with it by myself. If you have a desire, we can suggest you participate in another issue. Or at least in translation, after the API will be released.

**Comment by glauberm** [comment]
@neSpecc Thanks, I'll do that

**Comment by neSpecc** [comment]
Will be shipped with 2.18 release

**Comment by vincentdesmares** [comment]
@neSpecc Great news! I will try to help too. We just need the general guidelines on how the plugins should behave.

Also there are many small notifications to take in account (ex: the image plugin).

**Comment by ycs77** [comment]
I push a PR #1170 added i18n files feature 😄 
Let everyone contribute language files.

**Comment by neSpecc** [comment]
Resolved by #1181 
See https://editorjs.io/i18n
